Add standard set of vscode workspace setting files

rkeithhill opened the pull request:
Add an extensions.json file that will prompt folks to install extensions that are recommended for this workspace (c++, c#, powershell, cmake (editing syntax) and markdown linter.

Add settings.json to start to configure C# syntax, PoweShell trim trailing whitespace and configure the markdown lint extension.

Update launch.json to provide ability to debug PowerShell scripts.

BTW I'm hoping as more people on the team use the VSCode extension , it will help get more attention to fixing some lingering issues in the PS extension.  Dogfooding FTW!!

<!--

If you are a PowerShell Team member, please make sure you choose the Reviewer(s) and Assignee for your PR.
If you are not from the PowerShell Team, you can leave the fields blank and the Maintainers will choose them for you. If you are familiar with the team, feel free to mention some Reviewers yourself.

For more information about the roles of Reviewer and Assignee, refer to [CONTRIBUTING.md](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md).

-->

rkeithhill committed:
Add standard set of vscode workspace setting files

Add an extensions.json file that will prompt folks to install extensions that are recommended for this workspace (c++, c#, powershell, cmake (editing syntax) and markdown linter.

Add settings.json to start to configure C# syntax, PoweShell trim trailing whitespace and configure the markdown lint extension.

Update launch.json to provide ability to debug PowerShell scripts.

BTW I'm hoping as more people on the team use the VSCode extension , it will help get more attention to fixing some lingering issues in the PS extension.  Dogfooding FTW!!

rkeithhill commented:
BTW I should have added, the markdown linter integration into VSCode is really, really nice.  Much better to see issues while you are editing them than to wait for the Pester test to find them.  :-)

SteveL-MSFT commented on .vscode/settings.json:
```diff
@@ -0,0 +1,16 @@
+  "editor.tabSize": 4,
+  "editor.insertSpaces": true,
+  "markdownlint.config": {
+    "MD004": false,
```
can you add comment why these are being excluded?

rkeithhill committed:
Clean up grammar

SteveL-MSFT reviewed (approved):
LGTM

rkeithhill committed:
Instruct VSCode to ensure a final newline.

If this is not wanted for certain file types, we can restrict this option to only specified file types.

iSazonov commented on .vscode/extensions.json:
```diff
@@ -0,0 +1,11 @@
+{
+  // See http://go.microsoft.com/fwlink/?LinkId=827846
+  // for the documentation about the extensions.json format
+  "recommendations": [
```
GitHub don't like Json comments. I see https://stackoverflow.com/questions/244777/can-comments-be-used-in-json

Should we remove the comments or convert in another form?

rkeithhill replied:
Perhaps but it didn't stop the VSCode team from checking json with comments: https://github.com/Microsoft/vscode/blob/master/.vscode/extensions.json
https://github.com/Microsoft/vscode/blob/master/tslint.json

And that comment is the default comment added when you run the `Extensions: Configure Recommended Extensions` command in VSCode.

iSazonov replied:
If VS Code understands it, I'll merge.

iSazonov commented:
Reopen thePR to restart CI Appveyor

iSazonov commented:
@rkeithhill Many thanks for your contribution!
